command/git.go

```diff
@@ -193,9 +196,16 @@ func (c *gitFetchProject) ParseParams(params map[string]interface{}) error {
 	return nil
 }
 
+func (c *gitFetchProject) gitResetWithUnshallow(revision string) []string {
+	return []string{
+		fmt.Sprintf("git reset --hard %s && exit 0", revision),
+		fmt.Sprintf("$(git fetch --help | grep -q unshallow) && git fetch --unshallow || git fetch --depth %d", math.MaxInt32),
```

Comment:
What's the purpose of the first part?

Reply:
If git reset succeeds, then we should exit, since the revision exists. (It's in the 1000 cloned.) If, however, git reset did not succeed, then the revision is old. Instead of exiting, continue to the next line.